docs/docs/tutorial/creating-a-source-plugin/part-4/index.mdx

```diff
@@ -0,0 +1,866 @@
+    // highlight-next-line
+    import { NODE_TYPES, ERROR_CODES, CACHE_KEYS } from "./constants"
+
+    export const sourceNodes: GatsbyNode[`sourceNodes`] = async (gatsbyApi) => {
+      // highlight-next-line
+      const { reporter, cache } = gatsbyApi
+
+      // Rest of sourceNodes
+    }
+
+    // Rest of file
+    ```
+
+1. Generate a date timestamp just before the `fetchGraphQL` call and use `cache.set` to save the timestamp after the successful fetch. This way you're ensuring accurate timestamps that are only saved when a successful data call happened.
+
+    ```ts:title=plugin/src/source-nodes.ts
+    // Imports
+
+    export const sourceNodes: GatsbyNode[`sourceNodes`] = async (gatsbyApi) => {
+      // Rest of sourceNodes
+
+      // highlight-next-line
+      const lastFetchedDateCurrent = Date.now()
+
+      const { data, errors } = await fetchGraphQL<IApiResponse>(/* Query */)
+
+      if (errors) {
+        sourcingTimer.panicOnBuild({
+          id: ERROR_CODES.GraphQLSourcing,
+          context: {
+            sourceMessage: `Sourcing from the GraphQL API failed`,
+            graphqlError: errors[0].message,
+          },
+        })
+      }
+
+      // highlight-next-line
+      await cache.set(CACHE_KEYS.Timestamp, lastFetchedDateCurrent)
+
+      // Rest of sourceNodes
+    }
+
+    // Rest of file
+    ```
+
+    <Announcement>
+
+    `Date.now()` returns the number of milliseconds elapsed since the epoch. If you need another format, read the [Date.now() documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/now).
+
+    </Announcement>
+
+1. The next step is to use `cache.get` to retrieve the timestamp from the cache before the `fetchGraphQL` call. Since you won't use it for delta fetching in this tutorial, output the information in the `verbose` mode of Gatsby with `reporter.verbose`:
+
+    ```ts:title=plugin/src/source-nodes.ts
+    // Imports
+
+    export const sourceNodes: GatsbyNode[`sourceNodes`] = async (gatsbyApi) => {
+      // Rest of sourceNodes
+
+      // highlight-next-line
+      const lastFetchedDate: number = await cache.get(CACHE_KEYS.Timestamp)
+      const lastFetchedDateCurrent = Date.now()
+
+      // highlight-next-line
+      reporter.verbose(`[plugin] Last fetched date: ${lastFetchedDate}`)
+
+      // Rest of sourceNodes
+    }
+
+    // Rest of file
+    ```
+
+1. Time to test your changes! Restart the `develop:site` script but this time with the `--verbose` flag enabled:
+
+    ```shell
+    yarn develop:site --verbose
+    ```
+
+    The output got a lot busier! You should see a new log:
+
+    ```shell
+    verbose [plugin] Last fetched date: undefined
+    ```
+
+    You're seeing this because on this first run the `lastFetcheDate` didn't exist yet and thus `undefined` is returned from the `cache.get` call. Stop the development server and restart the script. You now should see a timestamp defined that looks something like this:
```

Comment:
Small typo:
```suggestion
    You're seeing this because on this first run the `lastFetchedDate` didn't exist yet and thus `undefined` is returned from the `cache.get` call. Stop the development server and restart the script. You now should see a timestamp defined that looks something like this:
```